### DISCUSSION TITLE
Support for Full-stack TLS?

### DISCUSSION THREAD

**Discussion by lewijw** [apache/airflow#34318] [Q&A]
I'm wanting to verify that Full-stack TLS (as defined below) is not currently supported by Airflow.  Specifically, it does not appear that Airflow can be deployed into k8s so that all communication between pods and nodes is encrypted.

I am also wondering if a PR was submitted to support it, how likely would it be accepted?

Full-stack TLS defined:  In the SAS documentation, Full-stack TLS is defined as:

https://go.documentation.sas.com/doc/en/sasadmincdc/v_043/calencryptmotion/p07i4iwkl3huj3n16jan24mw8apq.htm#n0oshbneft6071n1mbr2rvwacatv

Full-stack TLS includes Front-door TLS (the external entry points to the SAS Viya platform software) as well as encrypting all network traffic between back-end services. All network traffic is encrypted. Full-stack includes the entry points outside the Kubernetes cluster (the front-door servers are the ingress controller, SAS/CONNECT, CAS) and the back-end services that are internal to the Kubernetes cluster (SAS Viya microservices that communicate only with other services that are internal to the cluster). In order to encrypt back-end services, the ingress controller must re-encrypt traffic before forwarding it to the back-end SAS applications. SAS servers must use TLS when communicating directly with each other.

There is further information and a diagram here:

https://communities.sas.com/t5/SAS-Communities-Library/SAS-Viya-2021-1-3-TLS-Changes/ta-p/754712

**Comment by potiuk** [comment]
It has nothing to do with Airflow. This is a deployment issue. For example you could deploy Airflow with ISTIO to achieve what you want. But It's up to you how you deploy Airflow. Airflow as such is a collection of components and how you deploy it - is up to your K8s resources and implementation and you can configure any TLS scheme you want there, but it's outside of the domain of Airflow as a product.

As long as you deploy it with full TLS deployment you can use Airflow in full-TLS mode. But this is not a turn-key feature, you need to build the right deployment for Airflow to do it.
